ironfish/src/mining/minersFeeCache.ts

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import LRU from 'blru'
+import { IronfishNode } from '../node'
+import { Transaction } from '../primitives/transaction'
+import { SpendingAccount } from '../wallet'
+
+export class MinersFeeCache {
+  private readonly node: IronfishNode
+  private readonly cache: LRU<string, Promise<Transaction>> = new LRU<
+    string,
+    Promise<Transaction>
+  >(5)
```

Comment:
Does it make sense to only cache one empty miners fee transaction at a time? We'd only need to replace it when the node successfully mines a block or when the chain height changes the miners reward

Reply:
Discussed offline; this could be a followup optimization, but shouldn't get in the way of the overall goal of this PR